Python/compile.c

```diff
@@ -4642,6 +4642,9 @@ compiler_comprehension(struct compiler *c, expr_ty e, int type,
     qualname = c->u->u_qualname;
     Py_INCREF(qualname);
     compiler_exit_scope(c);
+    if ((c->c_flags->cf_flags & PyCF_ALLOW_TOP_LEVEL_AWAIT) && is_async_generator){
```

Comment:
Why not:

```suggestion
    if (top_level_await && is_async_generator){
```

you only want this to trigger on module-scope, no?

Reply:
Yeah sorry I was confused because of enter and exit scope.